ada_feeding_actionselect/ada_feeding_actionselect/adapters/context_posthoc_adapter.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+This module defines an abstract class (and defaults) for generating
+context and posthoc vectors from input data.
+"""
+
+# Standard imports
+from abc import ABC, abstractmethod
+
+# Third-party imports
+import numpy as np
+import numpy.typing as npt
+
+# Local imports
+from ada_feeding_msgs.msg import Mask
+
+
+class ContextAdapter(ABC):
+    """
+    An interface to translate a visual Mask to a context vector.
+    """
+
+    def __init__(self):
+        """
+        Default self properties
+        """
+        self.need_rgb = False
+        self.need_depth = False
+
+    @property
+    @abstractmethod
+    def dim(self) -> int:
+        """
+        Property defining the size of the context vector
+        """
+        raise NotImplementedError("dimension not implemented")
+
+    @abstractmethod
+    def get_context(
+        self, mask: Mask, image: npt.NDArray, depth: npt.NDArray
```

Comment:
```suggestion
        self, mask: Mask, image: Optional[npt.NDArray], depth: Optional[npt.NDArray]
```

You also need to import `Optional` from `typing` above